### PR TITLE
[TCH-10] add get product service

### PR DESCRIPTION
**Pull request by imwillim**

### CODE REVIEW

File: spec/services/get_product_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  let!(:category) { create(:category) }
+  let!(:sizes) { create_list(:size, 2, product: second_product) }
+  let!(:toppings) { create_list(:topping, 2, product: second_product) }
+  let!(:tag) { create(:tag, product: second_product) }
+  let!(:first_product) { create(:product, category:, name: 'Cà phê') }
+  let!(:second_product) { create(:product, id: product_id, category:, name: 'Trà sữa') }
+
+  describe '#validate' do
+    context 'when product_id is nil' do
```
Comment (bmhuyquoc104) [inline spec/services/get_product_service_spec.rb:18]: how about case when there is no product with given product_id exists in db ?

File: spec/factories/products.rb [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 FactoryBot.define do
   factory :product do
```
Comment (bmhuyquoc104) [inline spec/factories/products.rb:4]: should take a look at gem faker

File: spec/services/get_product_service_spec.rb (hunk 1 above)
Comment (imwillim) [reply spec/services/get_product_service_spec.rb:18]: Added that case, pls check again

File: spec/factories/products.rb (hunk 2 above)
Comment (imwillim) [reply spec/factories/products.rb:4]: Added gem faker for factory

**Review by bmhuyquoc104 — APPROVED** [review]

**Pull request merged**